src/main/scala/edu/berkeley/ce/rockslicing/JointGenerator.scala

```diff
@@ -0,0 +1,221 @@
+      Array(lowerLeftCorner(0) + deltaX, lowerLeftCorner(1), lowerLeftCorner(2)),
+      Array(lowerLeftCorner(0) + deltaX, lowerLeftCorner(1) + deltaY, lowerLeftCorner(2)),
+      Array(lowerLeftCorner(0), lowerLeftCorner(1) + deltaY, lowerLeftCorner(2))
+    )
+
+    val upperCorners = Array(
+      Array(lowerCorners(2)(0), lowerCorners(2)(1), lowerCorners(2)(2) + deltaZ),
+      Array(lowerCorners(3)(0), lowerCorners(3)(1), lowerCorners(3)(2) + deltaZ),
+      Array(lowerCorners(0)(0), lowerCorners(0)(1), lowerCorners(0)(2) + deltaZ),
+      Array(lowerCorners(1)(0), lowerCorners(1)(1), lowerCorners(1)(2) + deltaZ)
+    )
+
+    // Diagonal vectors of bounding box
+    val diagonalVectors = upperCorners.zip(lowerCorners) map { case (upper, lower) =>
+        linalg.normalize(DenseVector[Double](upper(0) - lower(0), upper(1) - lower(1), upper(2) - lower(2)))
+    }
+
+    jointTuple map { case (joint, jointData) =>
+      if (jointData(3) == 100) {
+        // Calculate dot product of each diagonal vector with joint normal
+        val diagDotProducts = diagonalVectors map { diagonal =>
+          math.abs(diagonal dot DenseVector[Double](joint.a, joint.b, joint.c))
+        }
+        // Use diagonal that is most parallel to joint normal - this ensures that joints generated will populate
+        // entire bounding box
+        val bestDiagonal = diagonalVectors(diagDotProducts.indexOf(diagDotProducts.max))
+        val diagonalIndex = diagDotProducts.indexOf(diagDotProducts.max)
+        val initialCorner = lowerCorners(diagonalIndex)
+        val terminalCorner = upperCorners(diagonalIndex)
+        makePersistentJointSet(joint, jointData, bestDiagonal, initialCorner, terminalCorner, Seq.empty[Joint])
+      } else {
+        makeNonPersistentJointSet(joint, jointData)
+      }
+    }
+  }
+
+  /**
+    * Generates a sequence of joints spanning the entire bounding box. The sequence of joints represents
+    * a single joint set.
+    *
+    * @param joint Joint that is representative of the joint set - properties of the joint are the mean values.
+    * @param jointData Properties of the joint set - strike, dip, joint spacing, persistence, phi, cohesion and
+    *                  optional stochastic parameters
+    * @param diagonalVector Diagonal vector of bounding box along which joints will be generated
+    * @param center Center of joint
+    * @param terminationPoint Point beyond which no further joints should be generated
+    * @param jointSet Seq of joints representing the joint set. This will be an empty Seq when the function is
+    *                 first called.
+    * @return Seq of joints representing the joint set
+    */
+  @tailrec
+  private def makePersistentJointSet(joint: Joint, jointData: Array[Double], diagonalVector: DenseVector[Double],
+                                     center: Array[Double], terminationPoint: Array[Double],
+                                     jointSet: Seq[Joint]): Seq[Joint] = {
```

Comment:
How about giving `jointSet` a default value of `Seq.empty[Joint]` so that we can omit it on the initial call?